### ISSUE TITLE
replacing @DragDropContext with @DndProvider generates error

### ISSUE THREAD

**Issue by HenrikBechmann** [react-dnd/react-dnd#1424]
**Describe the bug**
`error TS1238: Unable to resolve signature of class decorator when called as an expression.`
**To Reproduce**
replace `@DragDropContext` decorator with `@DndProvider` decorator
**Expected behavior**
no error, as with `@DragDropContext`
**Screenshots**

**Desktop (please complete the following information):**
 - OS: [e.g. iOS] MacOS version 10.14.5
 - Browser [e.g. chrome, safari] Chrome
 - Version [e.g. 22] Version 74.0.3729.169 (Official Build) (64-bit)

**Additional context**

**Comment by darthtrevino** [comment]
DndProvider is not a decorator, it's meant to be put into your React tree

**Comment by HenrikBechmann** [comment]
OK, so I put this in the tree:

`<DndProvider backend = {isMobile?DnDTouchBackend as any:DnDHtml5Backend}>`

(`as any` is required to get past the type checking)
 
... and I get this runtime error:

`HTML5Backend.js?54da:296 Uncaught Error: Cannot have two HTML5 backends at the same time.`

If I just do

`<DndProvider backend = {DnDTouchBackend}>`

I get this compile error:

`error TS2322: Type '{ children: Element[]; backend: (optionsOrManager: DragDropManager<any> | Opts) => TouchBackend | ((manager: DragDropManager<any>) => TouchBackend); }' is not assignable to type '(IntrinsicAttributes & { manager: DragDropManager<any>; } & { children?: ReactNode; }) | (IntrinsicAttributes & { backend: BackendFactory; context?: any; debugMode?: boolean; } & { ...; })'.
  Type '{ children: Element[]; backend: (optionsOrManager: DragDropManager<any> | Opts) => TouchBackend | ((manager: DragDropManager<any>) => TouchBackend); }' is not assignable to type '{ backend: BackendFactory; context?: any; debugMode?: boolean; }'.
    Types of property 'backend' are incompatible.
      Type '(optionsOrManager: DragDropManager<any> | Opts) => TouchBackend | ((manager: DragDropManager<any>) => TouchBackend)' is not assignable to type 'BackendFactory'.
        Type 'TouchBackend | ((manager: DragDropManager<any>) => TouchBackend)' is not assignable to type 'Backend'.
          Type '(manager: DragDropManager<any>) => TouchBackend' is missing the following properties from type 'Backend': setup, teardown, connectDragSource, connectDragPreview, connectDropTarget`

If I do 

`import DnDHtml5Backend from 'react-dnd-html5-backend'`
...
`<DndProvider backend = {DnDHtml5Backend}>`

... as specified in the usage documentation http://react-dnd.github.io/react-dnd/docs/api/dnd-provider#usage

I also get the runtime error

`HTML5Backend.js?54da:296 Uncaught Error: Cannot have two HTML5 backends at the same time.`

Any suggestions?

**Comment by darthtrevino** [comment]
That should be fixed in 8.0.3, which was cut just a few minutes ago

**Comment by HenrikBechmann** [comment]
Nice.

Now I have

```
import { DndProvider } from 'react-dnd'

import DnDTouchBackend from 'react-dnd-touch-backend'

import DnDHtml5Backend from 'react-dnd-html5-backend'

import application from '../services/application'

let isMobile = application.properties.ismobile

let DnDBackend = isMobile?DnDTouchBackend:DnDHtml5Backend
```
and 

`<DndProvider backend = {DnDBackend as any}>` 

in the tree, and it works on both desktop and mobile.

Thanks!!

- Henrik
